src/AWS/Types.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module AWS.Types where
+
+import Data.Aeson (FromJSON, withText)
+import Data.Text
+import GHC.Generics
+
+data ARN = ARN {
```

Comment:
This is the `ARN` format for API Gateway, but I don't think it would work other models e.g. kinesis.

Reply:
right! I'll make it more generic

Reply:
Now it's more generic, but i'm unhappy with the decoding, I'll write a parser tomorrow

Reply:
I don't think we have to go too deep on the models in this repo. We can customize them as necessary at the call site. In general, I think I'd prefer to close this PR and see if we can code gen all models in a very basic way e.g. `arn :: Text`.